java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java

```diff
@@ -1396,7 +1396,7 @@ public void nextVector(ColumnVector previousVector,
                            FilterContext filterContext) throws IOException {
       if (inBytesColVector == null) {
         // Allocate column vector for file; cast column vector for reader.
-        inBytesColVector = new BytesColumnVector();
+        inBytesColVector = new BytesColumnVector(batchSize);
```

Comment:
@pgaref . Could you add a test coverage for this change? It seems that we are missing `Varchar` to `Binary` conversion test case.

Reply:
Good catch @dongjoon-hyun ! test added